src/components/asset-characteristics/asset-characteristics.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+
+import { locale } from "../../services";
+
+import { AssetCharacteristics } from "@mtfh/common/lib/api/asset/v1";
+import { SummaryList, SummaryListItem } from "@mtfh/common/lib/components";
+
+interface IPropSpecProps {
+  assetCharacteristics: AssetCharacteristics;
+}
+
+type PropSpecProp = {
+  label: string;
+  value: string | number | null;
+};
+
+export const PropertySpecification = ({
+  assetCharacteristics,
+}: IPropSpecProps): JSX.Element => {
+  // Create an object of keys, labels, and values to loop through
+  const propSpecProps: {
+    [key: string]: PropSpecProp;
+  } = {
+    numberOfBedrooms: {
+      label: locale.assetCharacteristics.numberOfBedroomsLabel,
+      value: assetCharacteristics.numberOfBedrooms,
+    },
+    numberOfLifts: {
+      label: locale.assetCharacteristics.numberOfLiftsLabel,
+      value: assetCharacteristics.numberOfLifts,
+    },
+    numberOfSingleBeds: {
+      label: locale.assetCharacteristics.numberOfSingleBedsLabel,
+      value: assetCharacteristics.numberOfSingleBeds,
+    },
+    numberOfDoubleBeds: {
+      label: locale.assetCharacteristics.numberOfDoubleBedsLabel,
+      value: assetCharacteristics.numberOfDoubleBeds,
+    },
+    numberOfLivingRooms: {
+      label: locale.assetCharacteristics.numberOfLivingRoomsLabel,
+      value: assetCharacteristics.numberOfLivingRooms,
+    },
+    numberOfFloors: {
+      label: locale.assetCharacteristics.numberOfFloorsLabel,
+      value: assetCharacteristics.numberOfFloors,
+    },
+    totalBlockFloors: {
+      label: locale.assetCharacteristics.totalBlockFloorsLabel,
+      value: assetCharacteristics.totalBlockFloors,
+    },
+    heating: {
+      label: locale.assetCharacteristics.heatingLabel,
+      value: assetCharacteristics.heating,
+    },
+    windowType: {
+      label: locale.assetCharacteristics.windowTypeLabel,
+      value: assetCharacteristics.windowType,
+    },
+    propertyFactor: {
+      label: locale.assetCharacteristics.propertyFactorLabel,
+      value: assetCharacteristics.propertyFactor,
+    },
+    yearConstructed: {
+      label: locale.assetCharacteristics.yearConstructedLabel,
+      value: assetCharacteristics.yearConstructed,
+    },
+    architecturalType: {
+      label: locale.assetCharacteristics.architecturalTypeLabel,
+      value: assetCharacteristics.architecturalType,
+    },
+  };
+
+  // Loop through props as SummaryListItems
+  return (
+    <aside className="mtfh-asset-charateristics">
+      <SummaryList overrides={[0.5, 0.5]} variant="base">
+        {Object.keys(propSpecProps)
+          // Sort the labels not null on top
+          .sort((label) => {
+            if (propSpecProps[label].value == null) return 1;
+            return -1;
+          })
```

Comment:
I know it's deliberate... I'm personally not too fond of sorting as I feel that a rigid information display structure gives familiarity & ease of use going forward because "as the user I don't have to guess where within the asset characteristics list, my wanted piece of data is".

But that's my opinion... I can also see the "clean look/easier to read" argument behind the "sort" decision. I'd say these are close to equal. I'd probably want to brush this design through a user or anyone non-dev really. Pretty much ask: "Do you like the screenshot of non-sorted list, or sorted list more?".

Reply:
**TLDR;**
I think data fields jumping up & down in position going from one record to another will end up being more of a nuisance than a help. It may be good to drop the sorting/not-sorting decision onto the users so they make the choice themselves.

Reply:
Sorting removed, as confirmed by Sean, thanks for the heads-up.